File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/settings/account/AccountSettingsUiState.kt
package com.hhp227.concafe.presentation.settings.account

import com.hhp227.concafe.domain.model.MyInfoFeed
import com.hhp227.concafe.domain.model.UserRole

data class AccountSettingsUiState(
    val isLoading: Boolean,
    val errorMessage: String?,
    val myInfoFeed: MyInfoFeed?,
    val nicknameInput: String,
    val isDeleteDialogVisible: Boolean,
    val deletePassword: String,
    val deletePasswordErrorMessage: String?,
    val isDeleteRequested: Boolean
) {
    val role: UserRole?
        get() = myInfoFeed?.user?.role

    companion object {
        fun empty(): AccountSettingsUiState {
            return AccountSettingsUiState(
                isLoading = true,
                errorMessage = null,
                myInfoFeed = null,
                nicknameInput = "",
                isDeleteDialogVisible = false,
                deletePassword = "",
                deletePasswordErrorMessage = null,
                isDeleteRequested = false
            )
        }
    }
}

File: iosApp/iosApp/Presentation/Settings/Account/AccountSettingsUiState.swift
//
//  AccountSettingsUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/14.
//

import Foundation
import Shared

struct AccountSettingsUiState {
    var isLoading: Bool
    var errorMessage: String?
    var myInfoFeed: Shared.MyInfoFeed?
    var nicknameInput: String
    var isDeleteDialogVisible: Bool
    var deleteConfirmation: String
    var isDeleteRequested: Bool

    var role: UserRole? {
        myInfoFeed?.user?.role
    }

    static let empty = AccountSettingsUiState(
        isLoading: true,
        errorMessage: nil,
        myInfoFeed: nil,
        nicknameInput: "",
        isDeleteDialogVisible: false,
        deleteConfirmation: "",
        isDeleteRequested: false
    )
}
